### DISCUSSION TITLE
Filters not cleared when switching between vertical tabs

### DISCUSSION THREAD

**Discussion by psolano** [microsoft-search/pnp-modern-search#3010] [Ideas]
Version used = v4

**Describe the bug**
My goal is having a Search Verticals at the top which groups documents from a Document Library.

![image](https://user-images.githubusercontent.com/4042204/110376620-09b00380-8019-11eb-872a-b4ea0f36a056.png)

I have 1 Search Verticals with two tabs, 1 Search Filters and 2 Search Results

**Steps to reproduce**
The issue I have is that if I filter my results in one of the Tabs using the Filters; when I switch to the other tab, the filters are not cleared and the filters are gone since its still using the selection from the previous tab.

![image](https://user-images.githubusercontent.com/4042204/110376747-2cdab300-8019-11eb-8c70-f66d98702562.png)

![image](https://user-images.githubusercontent.com/4042204/110376859-5267bc80-8019-11eb-80c0-9f1453c34cb9.png)

**Expected behavior**
When the user click on another Vertical Tab; it should clear all selection.
Another approach could be adding an option to connect the Search Filter to a Vertical Search and add multiple Search Filters which will hide based on the selected Vertical tab

![image](https://user-images.githubusercontent.com/4042204/110379093-fce0df00-801b-11eb-9e79-055d45b0fef9.png)

This is a similar issue to #788

**Comment by FranckyC** [comment]
Hi @psolano, this is the default behavior for filters and verticals and it shouldn't be categorized as a bug. For some scenarios, it makes sens to same keep filters switching sources. We can see how to connect filters and verticals more easily as they have no connections for now and you are not the first to ask ;)

**Comment by psolano** [comment]
@FranckyC don't get me wrong, this Web Part is awesome with lots of out of the box settings and great flexibility for end users, but, I think this is a bug. This use case is about sharing the same Source (Document Library), filters are the same, let say Process Owner and Process Manager; however, values for those filters is what changes when changing between Verticals. It works fine if I had one filter selected and the other Search Results has the same value as a filter.

As of now, the only option I have is changing the approach of using Verticals and have one single Search Results and having a filter (Organization) that simulates the Verticals.

FYI, the query I have in my Search Results are:

- {SearchTerms} Path:{SiteCollection} AND TerOrgFilter:ATD
- {SearchTerms} Path:{SiteCollection} AND TerOrgFilter:QMS

**Comment by wobba** [comment]
@FranckyC I think the default should be to reload/bind the filters on a vertical change - as that's what happens on re-query. So it's not about clearing, but about re-triggering/binding the data.

**Comment by bryllyg** [comment]
Adding a request to keep the ability to persist the filter selections if the filter option appears on multiple verticals/pages. We are using 3.x right now and the users are frustrated that the choices are flushed when they change verticals. (We can't move to 4.x for the results page right now, since it does not yet support synonyms)

**Comment by wobba** [comment]
If you look at web search or OOB Microsoft Search (modern or classic), filters are not kept between verticals. I see why it can be useful when content on verticals is similar, but it's a special case.

If we could pass the picked vertical config as a URL token in the vertical config, that could perhaps work.

**Comment by wobba** [comment]
If you have two result wp's and the filter props are not the same MP, then they work individually. If they have the same property as filter, then it's applied to both. Need to investigate more and figure out all wanted options to solve this properly.

**Comment by DavorSukeri** [comment]
@wobba did you found any solution to this?

We have now two customers that have been complaining about this feature (=filter selections persist when changing vertical).

I did found an easy workaround to this, but its not perfect:

Use only one Search results webpart and add this to the Query template {searchTerms} {verticals.value} and add the actual query template to the Vertical value. You can use one or more Filter webparts connected to different verticals, but use only one Search Results webpart. The actual search query is defined in the vertical value (in the Search Vertical webpart).

That way it behaves differently.

If I now select a filter value in one vertical and there are some results and then change to other vertical (which is connected to the same Results and Filter webpart) the filter value persists, but it still shows results from the second vertical even there is not any results that are tied to the filter value that comes from the first vertical. 

I have also included "Show result count" showing after the filter value name for it to make more sense for the users, so they can see that filter value that was selected from the first vertical which doesn't bring any results shows (0) count in the filter webpart. This helps them understand why it doesn't show any results.
